save-backend/src/main/kotlin/org/cqfn/save/backend/controllers/UsersDetailsController.kt

```diff
@@ -53,17 +54,24 @@ class UsersDetailsController(
         val userId = (authentication.details as AuthenticationDetails).id
         val response = if (user.id == userId) {
             userRepository.save(user.apply {
-                email = newUserInfo.email
-                company = newUserInfo.company
-                location = newUserInfo.location
-                gitHub = newUserInfo.gitHub
-                linkedin = newUserInfo.linkedin
-                twitter = newUserInfo.twitter
+                email = newUserInfo.email.getValueOrNull() ?: email
+                password = "{bcrypt}${BCrypt.hashpw(newUserInfo.password.getValueOrNull() ?: password, BCrypt.gensalt())}"
+                company = newUserInfo.company.getValueOrNull() ?: company
+                location = newUserInfo.location.getValueOrNull() ?: location
```

Comment:
Yeah, when I did this, I didn't think about information deletion functionality.

However, with previous variant we got the **more problems** - since for information updates we use the common function `updateUser`, we can rewrite **already existing values in DB** by nulls or empty strings, despite the fact, that we didn't intent to do this. That's how it can happen:

For example, we are changing information about  `email`, but the `fieldsMap` with info about `company`, `github` etc will be empty in our case, because we are located in other tab. 
Therefore even if change only email, we will erase all info from other tabs. Same logic distributes for all fields from different tabs

I didn't invent some fast and good solution how to fix it, so let's probably use this workaround for some time and open an separate issue on this subject

Reply:
Can be prefilled when loading the view fieldsMap object from UserInfo

Reply:
Right, thanks, added